test/common/pktio_common.h

```diff
@@ -0,0 +1,124 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2020-2025 Nokia
+ */
+
+#ifndef PKTIO_COMMON_H
+#define PKTIO_COMMON_H
+
+#include <odp_api.h>
+#include <odp/helper/odph_api.h>
+
+#include <stdio.h>
+#include <inttypes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Print pktio link info
+ *
+ * @param pktio        Pktio handle
+ * @param pktio_name   Pktio name
+ */
+static void pktio_common_print_link_info(odp_pktio_t pktio, const char *pktio_name)
+{
+	odp_pktio_link_info_t info;
+
+	if (odp_pktio_link_info(pktio, &info)) {
+		ODPH_ERR("Error (%s): Printing pktio link info not possible, "
+			 "information retrieval failed.\n",
+			 pktio_name);
+		return;
+	}
+
+	printf("pktio: %s\n", pktio_name);
+	printf("  autoneg       %s\n",
+	       (info.autoneg == ODP_PKTIO_LINK_AUTONEG_ON ? "on" :
+	       (info.autoneg == ODP_PKTIO_LINK_AUTONEG_OFF ? "off" : "unknown")));
+	printf("  duplex        %s\n",
+	       (info.duplex == ODP_PKTIO_LINK_DUPLEX_HALF ? "half" :
+	       (info.duplex == ODP_PKTIO_LINK_DUPLEX_FULL ? "full" : "unknown")));
+	printf("  media         %s\n", info.media);
+	printf("  pause_rx      %s\n",
+	       (info.pause_rx == ODP_PKTIO_LINK_PAUSE_ON ? "on" :
+	       (info.pause_rx == ODP_PKTIO_LINK_PAUSE_OFF ? "off" : "unknown")));
+	printf("  pause_tx      %s\n",
+	       (info.pause_tx == ODP_PKTIO_LINK_PAUSE_ON ? "on" :
+	       (info.pause_tx == ODP_PKTIO_LINK_PAUSE_OFF ? "off" : "unknown")));
+	printf("  speed(Mbit/s) %" PRIu32 "\n\n", info.speed);
+}
+
+/**
+ * Print multiple pktio link infos.
+ * Same as pktio_common_print_link_info() but takes arrays of pktio handles and names
+ * and prints info for each link.
+ * 
+ * @param pktios        Pktio handles
+ * @param pktio_names   Pktio names
+ * @param num_pktio     Number of pktio handles and names
+ */
+static void pktio_common_print_link_info_multi(odp_pktio_t pktios[],
+					       const char *pktio_names[],
+					       uint32_t num_pktio)
+{
+	for (uint32_t i = 0; i < num_pktio; i++)
+		pktio_common_print_link_info(pktios[i], pktio_names[i]);
+}
+
+/**
+ * Wait for pktio links to be up and return status
+ *
+ * @param pktios        Pktio handles
+ * @param pktio_names   Pktio names
+ * @param num_pktio     Number of pktio handles and names
+ * @param wait_sec      Seconds to wait for all links to be up
+ *
+ * @retval 0 on success
+ * @retval -1 on failure
+ */
+static int pktio_common_check_link_status_wait(odp_pktio_t pktios[],
+					      const char *pktio_names[],
+					      uint32_t num_pktio,
+					      uint32_t wait_sec)
+{
+	odp_pktio_t pktio;
+	const char *name;
+	int link_status;
+	uint32_t link_wait = 0; /**< Centiseconds waited */
+
+	for (uint32_t i = 0; wait_sec && i < num_pktio; i++) {
```

Comment:
Now zero wait_sec makes this function a no-op. I think it would be nicer to have the function behave the same way, apart from the wait duration, regardless of whether wait_sec is zero or not, IOW, check the link states even with zero wait time. The call sites can just not call this function if they do not want anything to be done (which might be the case, for retaining existing behaviour if nothing else, in some  of the perf tests that are going to use this.